source/s3_request.c

```diff
@@ -29,6 +33,8 @@ struct aws_s3_request *aws_s3_request_new(
     request->meta_request = aws_s3_meta_request_acquire(meta_request);
 
     request->request_tag = request_tag;
+    request->request_type = request_type;
+    request->operation_name = operation_name;
```

Comment:
debatable: I don't really like the idea of request doesn't own the string, it feels easy to mess up the lifetime.

Reply:
I just thought out something like: We can have the operation name only for default type, and use the request type for all other operations. So, that we can safe some allocation, lol

Reply:
whatever, I just made it an aws_string
simple
obvious